packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useUploadFiles.test.ts

```diff
@@ -0,0 +1,464 @@
+        endpoint: mockExternalObjectStoreUploadDestination.endpointUrl,
+        keyPrefixUUID: mockExternalObjectStoreUploadDestination.keyPrefixUUID,
+      })
+
+      expect(mockUseSynapseMultipartUpload().mutateAsync).not.toHaveBeenCalled()
+      expect(mockUseSynapseMultipartUpload().mutateAsync).not.toHaveBeenCalled()
+      expect(mockOnUploadComplete).toHaveBeenCalledOnce()
+      expect(mockOnUploadComplete).toHaveBeenCalledWith(
+        { file: file1 },
+        createdFileHandleId1,
+      )
+    })
+
+    expect(hook.current.state).toBe('COMPLETE')
+  })
+
+  test('upload multiple files into Synapse Storage via Synapse multipart upload', async () => {
+    const startUploadArgs: BaseFileUploadArgs[] = [
+      { file: file1 },
+      { file: file2 },
+    ]
+
+    const useSynapseMultipartUploadMockReturn = getUseMutationIdleMock()
+    useSynapseMultipartUploadMockReturn.mutateAsync.mockResolvedValueOnce({
+      fileHandleId: createdFileHandleId1,
+      fileName: file1.name,
+    })
+    useSynapseMultipartUploadMockReturn.mutateAsync.mockResolvedValueOnce({
+      fileHandleId: createdFileHandleId2,
+      fileName: file2.name,
+    })
+    mockUseSynapseMultipartUpload.mockReturnValue(
+      useSynapseMultipartUploadMockReturn,
+    )
+    mockUseDirectS3Upload.mockReturnValue(getUseMutationIdleMock())
+
+    const { result: hook } = renderHook()
+
+    await waitFor(() => {
+      expect(hook.current.state).toBe('WAITING')
+      expect(hook.current.startUpload).toBeDefined()
+    })
+
+    act(() => {
+      hook.current.startUpload(...startUploadArgs)
+    })
+
+    await waitFor(() => {
+      expect(
+        useSynapseMultipartUploadMockReturn.mutateAsync,
+      ).toHaveBeenCalledTimes(2)
+      expect(
+        useSynapseMultipartUploadMockReturn.mutateAsync,
+      ).toHaveBeenCalledWith({
+        fileName: file1.name,
+        blob: file1,
+        storageLocationId: SYNAPSE_STORAGE_LOCATION_ID,
+        contentType: 'text/plain',
+        progressCallback: expect.any(Function),
+        abortController: expect.any(AbortController),
+        onMd5Computed: expect.any(Function),
+      })
+      expect(
+        useSynapseMultipartUploadMockReturn.mutateAsync,
+      ).toHaveBeenCalledWith({
+        fileName: file2.name,
+        blob: file2,
+        storageLocationId: SYNAPSE_STORAGE_LOCATION_ID,
+        contentType: 'text/plain',
+        progressCallback: expect.any(Function),
+        abortController: expect.any(AbortController),
+        onMd5Computed: expect.any(Function),
+      })
+      expect(mockOnUploadComplete).toHaveBeenCalledTimes(2)
+      expect(mockOnUploadComplete).toHaveBeenCalledWith(
+        { file: file1 },
+        createdFileHandleId1,
+      )
+      expect(mockOnUploadComplete).toHaveBeenCalledWith(
+        { file: file2 },
+        createdFileHandleId2,
+      )
+    })
+
+    expect(hook.current.state).toBe('COMPLETE')
+  })
+
+  test('pause and resume an upload', async () => {
```

Comment:
This test and the remaining tests below it are basically the same as tests that were previously in `useUploadFileEntities.test.ts`, with entity logic removed